src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -369,13 +394,18 @@ where
     async fn process_bind_msg(&mut self, msg: FeBindMessage) -> PsqlResult<()> {
         let statement_name = cstr_to_str(&msg.statement_name).unwrap().to_string();
         // 1. Get statement.
+        trace!(
+            "(extended query)bind: get statement name: {}",
+            &statement_name
+        );
         let statement = if statement_name.is_empty() {
-            &self.unnamed_statement
+            self.unnamed_statement.as_ref().ok_or_else(|| {
+                PsqlError::BindError(IoError::new(ErrorKind::InvalidInput, "No statement found"))
+            })?
         } else {
-            // NOTE Error handle method may need to modified.
-            // Postgresql doc needs write ErrorResponse if name not found. We may revisit
-            // this part if needed.
-            self.named_statements.get(&statement_name).expect("statement_name managed by client_driver, hence assume statement name always valid.")
+            self.named_statements.get(&statement_name).ok_or_else(|| {
+                PsqlError::BindError(IoError::new(ErrorKind::InvalidInput, "No statement found"))
+            })?
```

Comment:
Actually we should refine this later. This is not relates to IoError. IoError usually relates to buffer read/write.